test/cases/tasks/cockroachdb_rake_test.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "active_record/tasks/database_tasks"
+require "active_record/connection_adapters/cockroachdb/database_tasks"
+
+module ActiveRecord
+  class CockroachDBStructureDumpTest < ActiveRecord::TestCase
+    def setup
+      @configuration = {
+        "adapter"  => "cockroachdb",
+        "database" => "my-app-db"
+      }
+      @filename = "/tmp/awesome-file.sql"
+      FileUtils.touch(@filename)
+    end
+
+    def teardown
+      FileUtils.rm_f(@filename)
+    end
+
+    def test_structure_dump
+      assert_equal "", File.read(@filename)
+      File.write(@filename, "NOT TODAY\n")
+
+      config = @configuration.dup
+      config["database"] = ARTest.config["connections"]["cockroachdb"]["arunit"]["database"]
+
+      assert_called(
+        ActiveRecord::SchemaDumper,
+        :ignore_tables,
+        returns: ["accounts", "articles"]
+      ) do
+        ActiveRecord::Tasks::DatabaseTasks.structure_dump(config, @filename)
+
+        read = File.read(@filename)
+      end
+
+      read = File.read(@filename)
+      refute read.include?("NOT TODAY"), "The dump file previous content was not overwritten"
+      assert read.include?("CREATE TABLE public.schema_migrations"), "No dump done"
```

Comment:
should the schema and type statements be tested here too?

Reply:
well that is a good question. The tests in the ar test suit are even simpler than that (just checking for a comment). I can add those as they don't cost much to add. But I'm not sure of how secure this is. As stated before, I think an external and complete dump tool would be necessary to ensure some robustness!

Reply:
very fair point. i think the basic check you've added will at least catch a few regressions, even if it doesn't test each edge case.